packages/spanner/system-test/spanner.js

```diff
@@ -1034,9 +1035,56 @@ var spanner = new Spanner(env);
 
             transaction.commit(function(err) {
               assert.ifError(err);
+              firstTransaction.commit(assert.ifError);
```

Comment:
Your understanding is correct. The second transaction does need to commit for the first one to fail. I see your point about async. But I think the code would be easier to understand if we did not break the usual flow by calling commit somewhere else as you are doing now. Maybe you can some other mechanism to communicate to the first callback that the second txn has been committed, some sort of semaphore.